kong/plugins/hmac-auth/access.lua

```diff
@@ -132,8 +132,12 @@ local function create_hash(request_uri, hmac_params)
     if not header_value then
       if header == "request-line" then
         -- request-line in hmac headers list
+        local http_version = kong.request.get_http_version()
+        if http_version then
```

Comment:
Can `kong.request.get_http_version` _not_ return an HTTP version? 

* If yes, then line 140 below would end up writing `HTTP/nil`
* If no, then you should be able to just change the format string to below to `"%s %s HTTP/%.01f"` and leave the PDK call inside the `fmt`

Reply:
Well, we return nil ["for unrecognized values"](https://github.com/openresty/lua-nginx-module#ngxreqhttp_version) ... when I read that imagined someone sending `GET /foo HTTP/potato` instead of `GET /foo HTTP/1.0` over a telnet, but now that you mention it, I am not so sure such a request would reach this point. Not sure what "unrecognized values" mean then.

I've digged a bit more, and in the C source code there seems to be an allowance for it returning `nil`. I could not find a test or an explanation about when can it actually be nil:

https://github.com/openresty/lua-nginx-module/blob/master/src/ngx_http_lua_headers.c#L65

Leaving that aside, there's also the `http/nil`. Should we perhaps just error if the http version is not recognized?

Reply:
After consulting @dndx, it seems that these "unrecognized values" are not for regular http access, but for things like timers. We can be pretty sure it will not raise a `nil`, and if it does, an `assert` would suffice.